pkg/arvo/sys/vane/clay.hoon

```diff
@@ -4374,16 +4352,209 @@
+        ==
+      ::
+      ++  sign-1
+        $~  [%b %wake ~]
+        $%  $:  %y
+                $%  [%init-clad ~]
+            ==  ==
+            $:  %a
+                $>  $?  %send
+                        %woot
+                    ==
+                gift:able:ames
+            ==
+            $:  %b
+                $%  $>(%wake gift:able:behn)
+                    $>(%writ gift:able)
+            ==  ==
+            $:  %c
+                $>  $?  %mere
+                        %note
+                        %writ
+                    ==
+                gift:able
+            ==
+            $:  %f
+                $>(%made gift:able:ford)
+            ==
+            $:  %j
+                [%public-keys =public-keys-result-1]
+            ==
+            $:  @tas
+                $>(%crud vane-task)
+        ==  ==
+      ::
+      +$  public-keys-result-1
+        $%  [%full points=(map ship point:able:jael)]
+            [%diff who=ship =diff:point:able:jael]
+        ==
+      ::
+      ++  upgrade-1
+        |=  ruf=raft-1
+        ^-  axle
+        :-  %2
+        %=    ruf
+            act
+          ?~  act.ruf
+            ~
+          :-  ~
+          %=    u.act.ruf
+              eval-data
+            ?-    -.eval-data.u.act.ruf
+                %commit
+              :-  %commit
+              %-  (upgrade-clad-1 ,[dome rang])
+              form.eval-data.u.act.ruf
+            ::
+                %merge
+              :-  %merge
+              %-  (upgrade-clad-1 ,[(set path) dome rang])
+              form.eval-data.u.act.ruf
+            ::
+                %mount
+              :-  %mount
+              %-  (upgrade-clad-1 ,[(pair term beam) (map path mime)])
+              form.eval-data.u.act.ruf
+            ==
+          ==
+        ::
+            hoy
+          ^-  (map ship rung)
+          %-  ~(urn by hoy.ruf)
+          |=  [her=ship rift rus=(map desk rede-1)]
+          ^-  rung
+          %-  ~(urn by rus)
+          |=  [syd=desk red=rede-1]
+          ^-  rede
+          ?~  ref.red
+            red
+          %=    red
+              u.ref
+            ^-  rind
+            %=    u.ref.red
+                pud
+              ^-  update-qeu
+              ?~  eval-data.pud.u.ref.red
+                pud.u.ref.red
+              %=    pud.u.ref.red
+                  form.u.eval-data
+                %-  (upgrade-clad-1 ,(unit [lim=@da dome rang]))
+                form.u.eval-data.pud.u.ref.red
+              ==
+            ::
+                pur
+              %-  ~(urn by pur.u.ref.red)
+              |=  [inx=@ud =rand form=(eval-form-1 ,cage)]
+              [rand ((upgrade-clad-1 ,cage) form)]
+            ==
+          ==
+        ==
+      ::
+      ++  upgrade-clad-1
+        |*  a=mold
+        |=  e-form=(eval-form-1 a)
+        :-  effects.e-form
+        ^+  *form:(clad ,a)
+        |=  cin=clad-input
+        ?:  ?=([%j %public-keys %breach *] sign.cin)
+          [~ ~ %wait ~]
```

Comment:
I expected a %fail here, not a %wait.  It looks like this will just ignore a breach notification.  Do we not need to kill transactions involving a foreign ship that breaches while the transaction is still ongoing?  Or is Clay's input from a foreign ship always atomic for a given transaction?

Reply:
There's no real way it should get a public-keys response, since unlike in tapps, all the moves are on a particular wire prefix and we only route responses on that wire.

The transactions do need to be killed, but that happens elsewhere, where we actually handle the breach notification.